Src/coffeescript/cql-execution/src/elm/nullological.coffee

```diff
@@ -20,11 +20,12 @@ module.exports.Coalesce = class Coalesce extends Expression
 
   exec: (ctx) ->
     for arg in @args
-      # Lists are treated differently
-      if arg.elements?
-        for item in arg.elements
-          result = item.execute(ctx)
-          if result? then return result
+      # A single List (ExpressionRef->List) as the argument is treated differently
+      if @args.length == 1 && @args[0].constructor.name in ['List', 'ExpressionRef']
+        list = arg.execute(ctx)
+        if list?
+          for item in list
+            if item? then return item
```

Comment:
So...  I probably wasn't specific enough in my comment.  I was using `ExpressionRef` as an example of something that might _execute_ to a List (as opposed to a List literal).  There are other things that could execute to a list as well.  For example, a retrieve:
```
Coalesce([Encounter]
```
or a union:
```
define ListA = List{null, null, 3}
define ListB = List{4, null, 6}
define Three = Coalesce(ListA union ListB)
```

So, I think the code should maybe be something more like:
```coffee
    for arg in @args
      result = arg.execute(ctx)
      # if a single arg, and it's a list, then it's expected to coalesce over that list
      if @args.length == 1 && Array.isArray(result)
        for item in result
          if item? then return item
      else
        if result? then return result
    null
```
I didn't try compiling, but hopefully that conveys the idea well enough.